src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs

```diff
@@ -820,13 +831,19 @@ private void Write(ConsoleColor foregroundColor, ConsoleColor backgroundColor, s
 
                 try
                 {
-                    this.WriteImpl(value, newLine);
+                    this.WriteImpl(value, newLine: false);
                 }
                 finally
                 {
                     RawUI.ForegroundColor = fg;
                     RawUI.BackgroundColor = bg;
                 }
+
+                if (newLine)
+                {
+                    // write the newline after resetting the colors so there isn't a buffer wide colored line when buffer scrolls
+                    this.WriteImpl(string.Empty, newLine: true);
+                }
```

Comment:
I wonder why we reset only for newline.
I believe we should think about every output as atomic operation - set all attributes at start and reset all attributes at end because in the general case we do not know who and what will be output to the console in the next moment.

Reply:
Even if you `Write-Host -nonewline`, the formatandoutput will eventually emit a newline before the prompt.

Reply:
I mean some sequential write operations in script (before prompt) .